### PR TITLE
Bug 1141356 - utils-xpc.js ls recursive skipped if pattern not matched

### PR DESCRIPTION
**Pull request by cctuan**
https://bugzilla.mozilla.org/show_bug.cgi?id=1141356

### CODE REVIEW

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=722c72f2f2eb)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=d0a2112c942d)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=e585ed2906ae)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=2e19aaf6b31a)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=c9216d13e4d0)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=c4f60673d933)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=ccd1a14fda81)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=af850b9aa71c)

File: build/multilocale.js [hunk 1]
```diff
@@ -536,7 +536,10 @@ function execute(options) {
     return;
   }
   var buildDirectoryFile = utils.getFile(webapp.buildDirectoryFilePath);
-  var files = utils.ls(buildDirectoryFile, true, /^tests?$/);
+  var EXCLUDED = new RegExp(webapp.buildDirectoryFilePath + '.*\/tests?');
```
Comment (RickyChien) [inline build/multilocale.js]: I prefer to use lower-case ```excluded``` rather than upper-case

File: build/webapp-optimize.js [hunk 2]
```diff
@@ -694,8 +694,11 @@ WebappOptimize.prototype.execute = function(config) {
 
   // remove excluded condition /^(shared|tests?)$/)
   var buildDirectoryFile = utils.getFile(this.webapp.buildDirectoryFilePath);
-  var files = utils.ls(buildDirectoryFile, true,
-    /^(shared|tests?)$/);
+  var EXCLUDED =
```
Comment (RickyChien) [inline build/webapp-optimize.js]: same as above

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=00d03a991c5f)

**Comment by cctuan** [comment]
https://treeherder.mozilla.org/#/jobs?repo=try&revision=3c6fdc05a606

**Comment by cctuan** [comment]
push to try again
https://treeherder.mozilla.org/#/jobs?repo=try&revision=2ab2f85583c4

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=f684695dd277)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=1ed753b60be4)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=bd3215165a90)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=7d4cd597c8bf)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=9f1e574841cd)

**Pull request closed without merging**